test-common/src/main/scala/org/scalajs/testcommon/Serializer.scala

```diff
@@ -0,0 +1,290 @@
+package org.scalajs.testcommon
+
+import sbt.testing._
+import java.io._
+
+private[scalajs] trait Serializer[T] {
+  def serialize(x: T, out: Serializer.SerializeState): Unit
+  def deserialize(in: Serializer.DeserializeState): T
+}
+
+private[scalajs] object Serializer extends SerializerPlatformExtensions {
+  /* Serialization states. Serialization is currently stateless, but in the
+   * future we might want to deduplicate things.
```

Comment:
OK that's not at all clear to me from the current comment. Could you add that explanation in the comment?